core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -230,12 +251,16 @@ class NewHadoopRDD[K, V](
     super.persist(storageLevel)
   }
 
-
-  def getConf: Configuration = confBroadcast.value.value
```

Comment:
`HadoopRDD.getConf()` currently returns a new Configuration instance on each call, so maybe it's fine to make this change; I'd be very surprised if anyone was relying on this confusing undocumented behavior.